src/braket/devices/device.py

```diff
@@ -104,3 +110,43 @@ def status(self) -> str:
             str: The status of this Device
         """
         return self._status
+
+    def _validate_device_noise_model_support(self) -> None:
+        noise_pragma_name_to_class = copy(one_prob_noise_map)
+        noise_pragma_name_to_class.update({"kraus": Noise.Kraus})
+
+        supported_noises = [
+            noise_pragma_name_to_class[pragma.split("braket_noise_")[-1]]
+            for pragma in (self.properties.action[DeviceActionType.OPENQASM].supportedPragmas)
+            if "braket_noise_" in pragma
+        ]
+
+        noise_operators = [noise_instr.noise for noise_instr in self._noise_model._instructions]
+        if not all([isinstance(noise, tuple(supported_noises)) for noise in noise_operators]):
+            raise ValueError(
+                f"{self.name} does not support noise simulation or the noise model includes noise "
+                + f"that is not supported by {self.name}."
+            )
+
+    def _apply_noise_model_to_circuit(
+        self, task_specification: Union[Circuit, Problem, Program, AnalogHamiltonianSimulation]
+    ) -> None:
+        if isinstance(task_specification, Circuit):
+            if any(
+                [
+                    isinstance(instruction.operator, Noise)
+                    for instruction in task_specification.instructions
+                ]
+            ):
+                warnings.warn(
+                    "The noise model of the device is applied to a circuit that already has noise"
+                    " instructions."
+                )
```

Comment:
No need to go through all instructions in the circuit; you can stop searching the moment you find a noise instruction.